src/modeler/AutoRest.Swagger/SwaggerModeler.cs

```diff
@@ -293,20 +293,13 @@ public static string GetMethodName(Operation operation)
                 throw new ArgumentNullException("operation");
             }
 
-            if (operation.OperationId == null)
-            {
-                return null;
-            }
-
-            if (operation.OperationId.IndexOf('_') == -1)
-            {
-                return operation.OperationId;
-            }
-
-            var parts = operation.OperationId.Split('_');
-            return parts[1];
+            return GetMethodNameFromOperationId(operation.OperationId);
         }
 
+        public static string GetMethodNameFromOperationId(string operationId) => 
+            (operationId?.IndexOf('_') != -1) ? operationId.Split('_')[1]: operationId;
```

Comment:
Note that for multiple underscores, neither picking the 2nd nor the last one makes sense. So yes, we do assume it and making this simplification will at least be crystal clear with that (the other one only looks smarter when it really isn't 😆 )